Source/santad/main.mm

```diff
@@ -56,13 +52,11 @@ static void SantaWatchdog(void *context) {
   // santad's usual RAM usage is between 5-50MB but can spike if lots of processes start at once.
   const int mem_warn_threshold = 250;
 
-  struct mach_task_basic_info info;
-  mach_msg_type_number_t task_info_count = MACH_TASK_BASIC_INFO_COUNT;
+  struct proc_taskinfo pti;
 
-  if (task_info(mach_task_self(), MACH_TASK_BASIC_INFO, (task_info_t)&info, &task_info_count) ==
-      KERN_SUCCESS) {
+  if (GetTaskInfo(&pti)) {
     // CPU
-    double total_time = (timeval_to_double(info.user_time) + timeval_to_double(info.system_time));
+    double total_time = (double)MachTimeToNanos(pti.pti_total_user + pti.pti_total_system);
```

Comment:
Do you need to divide by NSEC_PER_SEC here?

Reply:
Yep, good catch - divide by seconds on the next line so need the conversion.